R/grouped_epi_archive.R

```diff
@@ -53,41 +53,22 @@ grouped_epi_archive <-
     public = list(
       initialize = function(ungrouped, vars, drop) {
         if (inherits(ungrouped, "grouped_epi_archive")) {
-          Abort("`ungrouped` must not already be grouped (neither automatic regrouping nor nested grouping is supported).  Either use `group_by` with `.add=TRUE`, or `ungroup` first.",
+          cli_abort("`ungrouped` must not already be grouped (neither automatic regrouping nor nested grouping is supported).  Either use `group_by` with `.add=TRUE`, or `ungroup` first.",
             class = "epiprocess__grouped_epi_archive__ungrouped_arg_is_already_grouped",
             epiprocess__ungrouped_class = class(ungrouped),
             epiprocess__ungrouped_groups = groups(ungrouped)
           )
         }
-        if (!inherits(ungrouped, "epi_archive")) {
-          Abort("`ungrouped` must be an epi_archive",
-            class = "epiprocess__grouped_epi_archive__ungrouped_arg_is_not_epi_archive",
-            epiprocess__ungrouped_class = class(ungrouped)
-          )
-        }
-        if (!is.character(vars)) {
-          Abort("`vars` must be a character vector (any tidyselection should have already occurred in a helper method).",
-            class = "epiprocess__grouped_epi_archive__vars_is_not_chr",
-            epiprocess__vars_class = class(vars),
-            epiprocess__vars_type = typeof(vars)
-          )
-        }
-        if (!all(vars %in% names(ungrouped$DT))) {
-          Abort("`vars` must be selected from the names of columns of `ungrouped$DT`",
-            class = "epiprocess__grouped_epi_archive__vars_contains_invalid_entries",
-            epiprocess__vars = vars,
-            epiprocess__DT_names = names(ungrouped$DT)
-          )
-        }
+        assert_class(ungrouped, "epi_archive")
+        assert_character(vars)
+        if (!test_subset(vars, names(ungrouped$DT))) cli_abort(
+          "All grouping variables `vars` must be present in the data.",
+        )
         if ("version" %in% vars) {
-          Abort("`version` has a special interpretation and cannot be used by itself as a grouping variable")
-        }
-        if (!rlang::is_bool(drop)) {
-          Abort("`drop` must be a Boolean",
-            class = "epiprocess__grouped_epi_archive__drop_is_not_bool",
-            epiprocess__drop = drop
-          )
+          cli_abort("`version` has a special interpretation and cannot be used by itself as a grouping variable")
         }
+        assert_logical(drop)
```

Comment:
```R
r$> assert_logical(5, len = 1)
Error: Assertion on '5' failed: Must be of type 'logical', not 'double'.
```

Reply:
Do you feel like the previous one was clearer? Is it the `rlang::last_trace()` message that you wish was there?